pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -38,8 +40,43 @@ import (
 // redisOpTimeout is the default operation timeout for the Redis proxy filter.
 var redisOpTimeout = 5 * time.Second
 
+// ListenerClass defines the class of the listener
+type telemetryMode int
+
+const (
+	statsAndMetadataExchange telemetryMode = iota
```

Comment:
we will want MX for logs and traces potentially too, right?  for instance, i may want to log client workload name or similar from the server. does that impact this `telemetryMode` calculation?

Reply:
I did not know we use MX for those? If so, it seems like the current EnvoyFilter implementation is broken since MX is only applied when telemetry.v2.enabled=true which is not related to logging or traces?

Reply:
We need it for stackdriver log.